nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,467 @@
+        ]
+        bagel_pipeline_df = bagel_pipeline_df[
+            bagel_pipeline_df[bagel.col_status] == STATUS_SUCCESS
+        ]
+
+        bagel_pipeline_df = bagel_pipeline_df.pivot(
+            index=[bagel.col_participant_id, bagel.col_session_id],
+            columns=self.col_pipeline,
+            values=bagel.col_status,
+        )
+
+        bagel_status_df = bagel_pipeline_df.groupby([bagel.col_session_id]).count()
+
+        self.logger.debug(f"bagel_status_df: {bagel_status_df}")
+
+        status_df = pd.concat([status_df, bagel_status_df], axis=1)
+
+        bagel_cols = list(status_df.columns[status_df.columns.str.contains("\n")])
+        return status_df, bagel_cols
+
+    def _add_manifest_rewards(
+        self, status_df: pd.DataFrame, status_col_dict: dict, reward_col_name: str
+    ) -> pd.DataFrame:
+        """Add manifest rewards column to the status DataFrame."""
+        manifest_col = status_col_dict["manifest"]
+        doughnut_cols = status_col_dict["doughnut"]
+        bagel_cols = status_col_dict["bagel"]
+
+        status_count_cols = [manifest_col] + doughnut_cols + bagel_cols
+        status_df[status_count_cols] = (
+            status_df[status_count_cols].fillna(0).astype(int)
+        )
+
+        for row_obj in status_df.iterrows():
+            row = row_obj[1]
+            reward_emojis = []
+            # clipboard emoji if some task (curation or processing) has been
+            # performed on participants in manifest else sad cat emoji
+            if row[manifest_col] > 0:
+                if (row[doughnut_cols] == 0).all() & (row[bagel_cols] == 0).all():
+                    reward_emojis.append(sad_cat_emoji)
+                else:
+                    reward_emojis.append(clipboard_emoji)
+
+            status_df.loc[row_obj[0], reward_col_name] = " ".join(reward_emojis)
+
+        return status_df
+
+    def _add_doughnut_rewards(
+        self, status_df: pd.DataFrame, status_col_dict: dict, reward_col_name: str
+    ) -> pd.DataFrame:
+        """Add doughnut rewards column to the status DataFrame."""
+        manifest_col = status_col_dict["manifest"]
+        doughnut_cols = status_col_dict["doughnut"]
+        bids_col = status_col_dict["bids"]
+        bagel_cols = status_col_dict["bagel"]
+
+        status_count_cols = [manifest_col] + doughnut_cols + bagel_cols
+        status_df[status_count_cols] = (
+            status_df[status_count_cols].fillna(0).astype(int)
+        )
+
+        for row_obj in status_df.iterrows():
+            row = row_obj[1]
+            reward_emojis = []
+            # doughnut emoji if at least one participant has been bidsified
+            # can't rely on other columns if we begin with BIDS dataset
+            if (row[bids_col] > 0) & (row[bids_col] <= row[manifest_col]):
+                reward_emojis.append(doughnut_emoji)
+            # sparkles emoji if all participants have been bidsified
+            if row[bids_col] == row[manifest_col]:
```

Comment:
Here if `bids_col` and `in_manifest` are both 0 they would get a sparkle too? Maybe we should make it so that if the manifest has nothing then no emojis are given? And then we should have a test case for that!